d2/src/test/java/com/linkedin/d2/xds/TestXdsClientImpl.java

```diff
@@ -443,6 +444,37 @@ public void testHandleD2URICollectionResponseWithRemoval()
     Assert.assertEquals(actualData.getURIMap(), D2_URI_MAP_UPDATE_WITH_DATA1.getURIMap());
   }
 
+  @Test
+  public void testWildCardResourceSubscription()
```

Comment:
Can we add testing to ensure that the wildcard subscription also subscribes to multiple different resources?

I think in this case we only test one service `SERVICE_RESOURCE_NAME` and one cluster `CLUSTER_RESOURCE_NAME`; I know we had a bug recently where we didn't handle multiple clusters correctly, but the tests missed it

Reply:
Actually, I thought we only need to verify the `WildcardResourceSubscriber` could subscribe the `Node` and `URIMap` resource.

The bug you mentioned actually always from INDIS server side, from the client side it should be good.